packages/ibm-products/src/components/CreateSidePanel/CreateSidePanel.tsx

```diff
@@ -26,6 +26,82 @@ const componentName = 'CreateSidePanel';
 
 // NOTE: the component SCSS is not imported here: it is rolled up separately.
 
+interface CreateSidePanelProps {
+  /**
+   * Provide an optional class to be applied to the containing node.
+   */
+  className?: string;
+  /**
+   * The description of the CreateSidePanel serves to provide more information about the form within the panel.
+   */
+  description?: ReactNode;
+  /**
+   * Specifies a boolean for disabling or enabling the primary button. This is important for form validation
+   * Returning `true` prevents the primary button from being clicked until required fields are completed.
+   */
+  disableSubmit?: boolean;
+  /**
+   * Specifies an optional field that provides a additional context for a form
+   */
+  formDescription?: ReactNode;
+
+  /**
+   * Specifies a required field that provides a title for a form
+   */
+  formTitle: ReactNode;
+
+  /**
+   * Unique identifier
+   */
+  id?: string;
+
+  /**
+   * Specifies an optional handler which is called when the CreateSidePanel
+   * is closed.
+   */
+  onRequestClose?(): void;
+  /**
+   * Specifies an optional handler which is called when the CreateSidePanel
+   * primary button is pressed.
+   */
+  onRequestSubmit?(): void;
+  /**
+   * Specifies whether the CreateSidePanel is open or not.
+   */
+  open: boolean;
+  /**
+   * Specifies the primary button's text in the modal.
+   */
+  primaryButtonText: string;
+  /**
+   * Specifies the secondary button's text in the modal.
+   */
+  secondaryButtonText: string;
+  /**
+   * This is the selector to the element that contains all of the page content that will shrink if the panel is a slide in.
+   * This prop is required when using the `slideIn` variant of the side panel.
+   */
+  selectorPageContent: string;
```

Comment:
Did not realize the `CreateSidePanel` defaults to `slideIn`, but probably makes sense for the use case?

I’d probably suggest editing the comment slightly since `slideIn` is a given in this case? 🤔 

```suggestion
   * This is the selector to the element that contains all of the page content that will shrink when the panel is a slide in.
   * This prop is required since create flows use the `slideIn` variant of the side panel.
   */
  selectorPageContent: string;
```

Reply:
^ just brainstorming. Probably still better to have a bit of the `slideIn` context?